exercises/practice/robot-simulator/robot_simulator_test.gd

```diff
@@ -0,0 +1,126 @@
+func same_field_values(robot_a, robot_b):
+	return robot_a.current_position == robot_b.current_position and robot_a.current_direction == robot_b.current_direction
+
+
+func test_create_robot_at_origin_facing_north(solution_script):
+	var robot_a = solution_script.RobotSimulator.new(Vector2i(0, 0), 'north')
```

Comment:
I'd like to suggest a similar architecture that I used in Luhn exercise:
* the solution script is the robot itself (no class)
* direction and position are the properties of the script (`@export var position : Vector2i` etc), and they are set like `solution_script.position = Vector2i(1, 1)`
* the `move()` method is called on the script itself (`solution_script.move()`)
* new direction and position can be read from the script to check the results 

In that case, the solution script would not need any classes or extra indents. This is a way Godot nodes usually work, with a custom node being e.g. an enemy in a game, a bullet etc. The game later creates instances of them and sets the properties (and calls methods) as described above.

We can even rename `solution_script` argument to `robot` in test methods, to make things more clear :) (in a normal Godot program, a scene would have a root node with a descriptive name, and it would be saved to a file with a similar name).

I think the comparison of results can, in this case, be simplified to something like:

```
return [[robot.position, robot.direction]. [Vector2i(1, 1), 'south']]
```

Although I would have to check if Godot handles comparisons of lists ...

I've noticed that you also use "private" variables (like `_position`). This is not a problem in my solution, this naming convention works great, and the "private" variables could simply not be exported (they would still be accessible, but that would be a nice hint not to access them directly).

What do you think? :)

Reply:
Sounds good. I’ll simplify the solution this week.